### PR TITLE
Added handling disallow_attester_search_for_domains OrgRule

### LINKED ISSUE #1203: implement disallow_attester_search_for_domains OrgRule
including wildcard `*` support meaning don't search on attester at all

### PR DESCRIPTION
**Pull request by DenBond7**
This PR added handling disallow_attester_search_for_domains OrgRule

close #1203

----------------------------------

**Tests** _(delete all except exactly one)_:
- Tests added or updated

--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [ ] addresses the issue it closes (if any)
- [ ] code is readable and understandable
- [ ] is accompanied with tests, or tests are not needed
- [ ] is free of vulnerabilities

### CODE REVIEW

File: FlowCrypt/src/main/java/com/flowcrypt/email/api/retrofit/response/model/OrgRules.kt [hunk 1]
```diff
@@ -184,9 +184,25 @@ data class OrgRules constructor(
...
+      return false
+    }
+
+    val disallowedDomains = disallowAttesterSearchForDomains ?: emptyList()
+    val userDomain = EmailUtil.getDomain(emailAddr)
+    if (userDomain.isEmpty()) {
+      throw IllegalStateException("Not a valid email $emailAddr")
+    }
+
+    return !disallowedDomains.contains(userDomain)
```
Comment (IvanPizhenko) [inline FlowCrypt/src/main/java/com/flowcrypt/email/api/retrofit/response/model/OrgRules.kt:197]: doesn't this require sort of case-insensitive comparison?

**Review by IvanPizhenko — COMMENTED** [review]
I have a question

File: FlowCrypt/src/main/java/com/flowcrypt/email/api/retrofit/response/model/OrgRules.kt (hunk 1 above)
Comment (DenBond7) [reply FlowCrypt/src/main/java/com/flowcrypt/email/api/retrofit/response/model/OrgRules.kt:197]: This realization is based on https://github.com/FlowCrypt/flowcrypt-browser/blob/48cb02a244b71a2acffbbc0fdb6ed4254baa0da3/extension/js/common/org-rules.ts#L170-L192.

Does `include` mean case-insensitive comparison?

Comment (IvanPizhenko) [reply FlowCrypt/src/main/java/com/flowcrypt/email/api/retrofit/response/model/OrgRules.kt:197]: In typescript we have:
```typescript
  public canLookupThisRecipientOnAttester = (emailAddr: string): boolean => {
    ....
    const userDomain = Str.getDomainFromEmailAddress(emailAddr);
    if (!userDomain) {
      throw new Error(`Not a valid email ${emailAddr}`);
    }
    return !disallowedDomains.includes(userDomain);
  }
```
and then:
```typescript
  public static getDomainFromEmailAddress = (emailAddr: string) => {
    // todo: parseEmail()?
    return emailAddr.toLowerCase().split('@')[1];
  }
```
So function for getting domain in the typescript casts it to lowercase and then if you configure everything in lowercase, all works.
In Kotlin that doesn't happen:
```kotlin
    // from EmailUtil.kt
    fun getDomain(email: String): String {
      return when {
        TextUtils.isEmpty(email) -> ""
        email.contains("@") -> email.substring(email.indexOf('@') + 1)
        else -> ""
      }
    }
```

Comment (DenBond7) [reply FlowCrypt/src/main/java/com/flowcrypt/email/api/retrofit/response/model/OrgRules.kt:197]: I see. Thank you. I will fix it.

**Comment by tomholub** [comment]
Well noticed, Ivan - thank you.

**Review by IvanPizhenko — APPROVED** [review]

**Pull request merged**